### PR TITLE
Added ADThermoDiffusion

### PR DESCRIPTION
**Pull request by tophmatthews**
Ref #12633

### CODE REVIEW

**Comment by tophmatthews** [comment]
@lindsayad Another for Soret.

**Comment by tophmatthews** [comment]
`SoretDiffusion` and `ThermoDiffusion` are similar, but slightly different. They can both be combined into something where the Soret coefficient is calculated elsewhere. I want the Soret diffusion calculated elsewhere, and only multiplied against the temperature variable. Neither `SoretDiffusion` nor `ThermoDiffusion` really do what I want, that's why I made this separate.

**Comment by gambka** [comment]
If you register your test object to MiscTestApp instead of MiscApp it shouldn't check for documentation for that object.  Then when testing the object you can include `allow_test_objects` in the tests file for that particular test.

**Comment by tophmatthews** [comment]
Thanks @gambka!

**Comment by tophmatthews** [comment]
@lindsayad How often is `devel` branch updated? I can go off of `next` if I have to though

**Comment by lindsayad** [comment]
`devel` is updated whenever all the tests on next (viewable at [moosebuild.org](moosebuild.org)) are green or orange (no reds allowed). Currently our "sync presentation" test is failing which is probably due to the same reason that all of our "sync to remote" tests on documentation tests are failing. So until we get that worked out, it's perhaps best to work off of next if there are things you need for your PR in next that are not in devel

**Comment by tophmatthews** [comment]
Since I was touching the gold file already, I thought I'd clean up the test at the same time.

File: modules/misc/test/tests/kernels/thermo_diffusion/tests [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+    requirement = 'AD thermo diffusion shall reproduce an analytical solution'
+    design = ""
+    issues = "#12633"
+    allow_test_objects = true
+  [../]
+  [./ad_thermo_diffusion_jacobian]
+    type = 'PetscJacobianTester'
+    input = 'ad_thermo_diffusion.i'
+    ratio_tol = 1e-7
+    difference_tol = 1e-3
```
Comment (tophmatthews) [inline modules/misc/test/tests/kernels/thermo_diffusion/tests]: @lindsayad I'm not sure why this had to be loosened so much, it seems like the values are already low enough to signify a good jacobian?

Comment (lindsayad) [reply modules/misc/test/tests/kernels/thermo_diffusion/tests]: `ratio_tol` is a more important indicator of the Jacobian accuracy. If you have "larger" Jacobian values (e.g. a large Young's modulus in solid mechanics simulations without any manual scaling), then this `difference_tol` in general will need to be larger

Comment (tophmatthews) [reply modules/misc/test/tests/kernels/thermo_diffusion/tests]: Okay, so if this is all okay, then this pr should be good to go for review

**Review by lindsayad — APPROVED** [review]
Good code and well-documented tests

**Pull request merged**